crates/chain/src/local_chain.rs

```diff
@@ -34,6 +34,20 @@ struct CPInner {
     prev: Option<Arc<CPInner>>,
 }
 
+impl PartialEq for CheckPoint {
+    fn eq(&self, other: &Self) -> bool {
```

Comment:
:wrench: https://doc.rust-lang.org/core/iter/trait.Iterator.html#method.eq

I think it can be implemented with this.

Reply:
Would that not cause a recursion though?

Reply:
Oh interesting point. We still want to map to the `BlockId` so the eq used in the iterator is just `BlockId`'s. That should be fine!

Reply:
Good point @ValuedMammal. I'll map the iterator to block id first.

Reply:
Updated to use `Iterator::eq`.